ts/components/Collapsible.svelte

```diff
@@ -0,0 +1,76 @@
+<!--
+Copyright: Ankitects Pty Ltd and contributors
+License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+-->
+<script lang="ts">
+    import { promiseWithResolver } from "../lib/promise";
+
+    export let id: string | undefined = undefined;
+    let className: string = "";
+    export { className as class };
+
+    export let collapsed = false;
+    let height = 0;
+
+    const [element, elementResolve] = promiseWithResolver<HTMLElement>();
+    let isCollapsed = false;
+
+    let style: string;
+    const setStyle = (el: HTMLElement) => {
+        height = el.clientHeight;
+        style = `--collapse-height: -${height}px; --duration: ${Math.sqrt(
+            height * 80,
+        )}ms`;
+    };
+
+    const transition = async (collapse: boolean) => {
+        const inner = await element;
+
+        isCollapsed = true;
+
+        if (collapse) {
+            setStyle(inner);
+
+            inner.addEventListener(
+                "transitionend",
+                () => {
+                    // DOM-manipulation is required here
+                    inner.setAttribute("hidden", "");
+                },
+                { once: true },
+            );
+        } else {
+            inner.removeAttribute("hidden");
+            // force repaint to enable transition
+            const repaint = inner.offsetHeight;
+            isCollapsed = false;
+        }
+    };
+
+    $: transition(collapsed);
+</script>
+
+<div {id} class="collapsible-container {className}">
+    <div
+        class="collapsible-inner"
+        class:is-collapsed={isCollapsed}
+        use:elementResolve
+        {style}
+    >
+        <slot />
+    </div>
+</div>
+
+<style lang="scss">
+    .collapsible-container {
+        position: relative;
+        overflow: hidden;
```

Comment:
The MathJax overlay getting constrained is an unintentional side-effect of this line, which means I'll just have to use a different method for the slide effect.

Reply:
I wouldn't be too fast here. The MathJax overlay being nested that deeply into the the HTML tree is the actual problem here. `WithFloating` works quite differently to that, by removing the component from the tree and adding it at the bottom of `document.body`:

<img width="1192" alt="Bildschirmfoto 2022-08-14 um 12 22 40" src="https://user-images.githubusercontent.com/7188844/184532697-3ad106ec-a945-41d0-81cc-aa386c05299b.png">

Reply:
Beccause the margin-top + clipping approach is more natural-looking than other solutions involving height or scaleY(), I decided to keep it and hide the overflow only during transition (https://github.com/ankitects/anki/pull/2002/commits/721ab0a23a7e208ea96f01939e9d6496c347255f).

Reply:
@hgiesel Sorry, the GitHub page I had open didn't update so I missed your comment.

Reply:
If you don't mind, I would leave the MathJax handle untouched in this PR because my recent changes don't make this an immediate issue anymore.

Reply:
No worries, I already started working on it: #2011